src/backend/d3d12/CommandAllocatorManager.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BACKEND_D3D12_COMMANDALLOCATORMANAGER_H_
+#define BACKEND_D3D12_COMMANDALLOCATORMANAGER_H_
+
+#include "d3d12_platform.h"
+
+#include "common/SerialQueue.h"
+
+#include <bitset>
+
+namespace backend {
+namespace d3d12 {
+
+    class Device;
+
+    class CommandAllocatorManager {
+        public:
+            CommandAllocatorManager(Device* device);
+
+            // A CommandAllocator that is reserved must be used before the next Device::Tick where the next serial has completed on the GPU
```

Comment:
nit: I can't parse the comment. The sentence seems to means

 - A reserved allocator must be used before only before the next Device::Tick
 - It will be reset when the GPU as completed work previous to that Tick (does it matter to the user of this interface?)